### ISSUE TITLE
Infinite loop in fn open

### ISSUE THREAD

**Issue by Ben-Lichtman** [mdsteele/rust-cfb#21]
```rust
        while current_difat_sector != consts::END_OF_CHAIN {
            // -- SNIP -- //
            difat_sector_ids.push(current_difat_sector);
            let mut sector = sectors.seek_to_sector(current_difat_sector)?;
            for _ in 0..(sector_len / size_of::<u32>() - 1) {
                // -- SNIP -- //
            }
            current_difat_sector = sector.read_u32::<LittleEndian>()?;
        }
```
In the above code, there is the possibility that the `read_u32` at the end of the loop returns the same index as has just been read. This results in a seek back to the same sector that has just been read and an infinite loop.

I'm unsure of the best way to solve this problem though - ie. If you simply check whether the new sector equals the current sector that won't prevent cycles of 2 or more steps, however it may be a good start in any case...

Another option might be to use a HashSet to detect duplicates of the `current_difat_sector` variable between loops and error out if one is found.

**Comment by Ben-Lichtman** [comment]
Same also applies to `current_dir_sector` later in the function

**Comment by mdsteele** [comment]
Thanks!  Seems like this should be fixed by PR #22.
